pygama/dsp/_processors/time_point_thresh.py

```diff
@@ -1,21 +1,59 @@
 import numpy as np
 from numba import guvectorize
+import math
+from pygama.dsp.errors import DSPFatal
 
-@guvectorize(["void(float32[:], float32, int32, int32[:])",
-              "void(float64[:], float64, int32, int32[:])",
-              "void(int32[:], int32, int32, int32[:])",
-              "void(int64[:], int64, int32, int32[:])"],
-             "(n),(),()->()", nopython=True, cache=True)
-def time_point_thresh(wf_in, threshold, tp_max, tp_out):
+@guvectorize(["void(float32[:], float32, float32, int32 ,float32[:])",
```

Comment:
It turns out that using float32s for time values will cause conflicts with time values passed as int32s. This happened to me using the output of argmax as t_start here. A solution is to use float64s for time values, since int32 can safely convert to float64